src/plugin.js

```diff
@@ -366,46 +323,41 @@ class VR extends Plugin {
 
     this.scene = new THREE.Scene();
 
-    this.videoTexture = new THREE.VideoTexture(this.getVideoEl_());
-
-    this.videoTexture.generateMipmaps = false;
-    this.videoTexture.minFilter = THREE.LinearFilter;
-    this.videoTexture.magFilter = THREE.LinearFilter;
-
-    // iOS and macOS HLS fix/hacks
+    // We opted to stop using a video texture on safari due to
+    // various bugs that exist when using it. This gives us worse performance
+    // but it will actually work on all recent version of safari. See
+    // the following issues for more info on this:
+    //
     // https://bugs.webkit.org/show_bug.cgi?id=163866#c3
-    // https://github.com/mrdoob/three.js/issues/9754
-    // On iOS with HLS, color space is wrong and texture is flipped on Y axis
-    // On macOS, just need to flip texture Y axis
-
-    if (isHLS() && videojs.browser.IS_ANY_SAFARI) {
-      this.log('Safari + iOS + HLS = flipY and colorspace hack');
-      this.videoTexture.format = THREE.RGBAFormat;
-      this.videoTexture.flipY = false;
-    } else if (isHLS() && videojs.browser.IS_SAFARI) {
-      this.log('Safari + HLS = flipY hack');
-      this.videoTexture.format = THREE.RGBFormat;
-      this.videoTexture.flipY = false;
-    } else {
-      this.videoTexture.format = THREE.RGBFormat;
-    }
+    // https://bugs.webkit.org/show_bug.cgi?id=179417
+    if (videojs.browser.IS_ANY_SAFARI && utils.isHLS(this.player_.currentSource().type)) {
```

Comment:
previously we had a hack to get video textures working here, this uses the canvas image texture instead as that will make ios 10 and 11 both work

Reply:
From my testing on an IPAD It doesn't perform bad enough that I think its a bad solution.